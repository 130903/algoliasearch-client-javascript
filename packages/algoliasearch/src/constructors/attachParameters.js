// @flow

export default function attachParameters(original: Object, extra: Object) {
  const methodNames = Object.keys(original);

  const augmentedMethods = methodNames.reduce(
    (methods, method) => ({
      ...methods,
      [method]: ({ ...args }) => original[method]({ ...extra, ...args }),
    }),
    {}
  );

  return augmentedMethods;
}
