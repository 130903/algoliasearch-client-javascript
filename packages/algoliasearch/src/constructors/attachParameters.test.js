import attachParameters from './attachParameters';

it('attaches an existing parameter', () => {
  const multipleFunctions = {
    someFn: ({ toAdd, host }) => ({ toAdd, host }),
    anotherUnused: () => {},
  };

  const augmentedFunctions = attachParameters(multipleFunctions, {
    toAdd: 'some extra new param',
  });

  expect(augmentedFunctions).toMatchSnapshot();

  expect(augmentedFunctions.someFn()).toMatchObject({
    host: undefined,
    toAdd: 'some extra new param',
  });

  expect(augmentedFunctions.someFn({ host: 'something' })).toMatchObject({
    host: 'something',
    toAdd: 'some extra new param',
  });

  expect(
    augmentedFunctions.someFn({ host: 'something', toAdd: 'override' })
  ).toMatchObject({
    host: 'something',
    toAdd: 'override',
  });
});
