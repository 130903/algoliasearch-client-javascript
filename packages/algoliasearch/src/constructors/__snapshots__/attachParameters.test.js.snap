// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attaches an existing parameter 1`] = `
Object {
  "anotherUnused": [Function],
  "someFn": [Function],
}
`;
